test/test_fx_reinplace_pass.py

```diff
@@ -0,0 +1,114 @@
+# Owner(s): ["module: functionalization"]
+import torch
+from torch.testing._internal.common_utils import TestCase, run_tests
+from torch.fx.passes.reinplace import reinplace
+from torch.fx.experimental.proxy_tensor import make_fx
+
+try:
+    from functorch.experimental import functionalize
+    HAS_FUNCTIONALIZATION = True
+except e:
+    HAS_FUNCTIONALIZATION = False
+
+class TestReinplacePass(TestCase):
+
+    def test_reinplace_basic(self):
+        # Basic test: the out-of-place add() call should be converted
+        # into add_()
+        def f(x):
+            a = x.clone()
+            b = a.add(1)
+            return b
+
+        inpt = torch.ones(2)
+        f2 = reinplace(make_fx(f)(inpt), inpt)
```

Comment:
If we do https://github.com/pytorch/pytorch/issues/82044 it shouldn't be necessary to feed inputs into `reinplace` as it should be able to just rely on the fake tensors stashed on the FX IR itself.